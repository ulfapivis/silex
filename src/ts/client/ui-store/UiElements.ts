
export interface UiElements {
  fileExplorer: HTMLIFrameElement
  contextMenu: HTMLElement
  menu: HTMLElement
  breadCrumbs: HTMLElement
  pageTool: HTMLElement
  htmlEditor: HTMLElement
  cssEditor: HTMLElement
  jsEditor: HTMLElement
  settingsDialog: HTMLElement
  dashboard: HTMLElement
  propertyTool: HTMLElement
  textFormatBar: HTMLElement
  workspace: HTMLElement
  verticalSplitter: HTMLElement
}

// keep references to ui elements
let uiElements: UiElements
export function getUiElements(): UiElements {
  uiElements = uiElements || {
    fileExplorer: document.querySelector('#silex-file-explorer'),
    contextMenu: document.querySelector('.silex-context-menu'),
    menu: document.querySelector('.silex-menu'),
    breadCrumbs: document.querySelector('.silex-bread-crumbs'),
    pageTool: document.querySelector('.silex-page-tool'),
    htmlEditor: document.querySelector('.silex-html-editor'),
    cssEditor: document.querySelector('.silex-css-editor'),
    jsEditor: document.querySelector('.silex-js-editor'),
    settingsDialog: document.querySelector('.silex-settings-dialog'),
    dashboard: document.querySelector('.silex-dashboard'),
    propertyTool: document.querySelector('.silex-property-tool'),
    textFormatBar: document.querySelector('.silex-text-format-bar'),
    workspace: document.querySelector('.silex-workspace'),
    verticalSplitter: document.querySelector('.vertical-splitter'),
  } as UiElements
  return uiElements
}
