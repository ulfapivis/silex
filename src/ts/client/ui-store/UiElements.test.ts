import { getUiElements } from './UiElements'

beforeEach(() => {
  document.body.innerHTML = `
    <iframe src="about:blank" id="silex-stage-iframe"></iframe>
    <iframe src="about:blank" id="silex-file-explorer"></iframe>
    <div class="silex-context-menu"></div>
    <div class="silex-menu"></div>
    <div class="silex-bread-crumbs"></div>
    <div class="silex-page-tool"></div>
    <div class="silex-html-editor"></div>
    <div class="silex-css-editor"></div>
    <div class="silex-js-editor"></div>
    <div class="silex-settings-dialog"></div>
    <div class="silex-dashboard"></div>
    <div class="silex-property-tool"></div>
    <div class="silex-text-format-bar"></div>
    <div class="silex-workspace"></div>
    <div class="vertical-splitter"></div>
  `
})

test('get all ui elements', () => {
  expect(getUiElements()).toMatchSnapshot()
})
