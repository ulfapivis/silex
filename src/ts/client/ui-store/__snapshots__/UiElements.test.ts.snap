// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get all ui elements 1`] = `
Object {
  "breadCrumbs": <div
    class="silex-bread-crumbs"
  />,
  "contextMenu": <div
    class="silex-context-menu"
  />,
  "cssEditor": <div
    class="silex-css-editor"
  />,
  "dashboard": <div
    class="silex-dashboard"
  />,
  "fileExplorer": <iframe
    id="silex-file-explorer"
    src="about:blank"
  />,
  "htmlEditor": <div
    class="silex-html-editor"
  />,
  "jsEditor": <div
    class="silex-js-editor"
  />,
  "menu": <div
    class="silex-menu"
  />,
  "pageTool": <div
    class="silex-page-tool"
  />,
  "propertyTool": <div
    class="silex-property-tool"
  />,
  "settingsDialog": <div
    class="silex-settings-dialog"
  />,
  "textFormatBar": <div
    class="silex-text-format-bar"
  />,
  "verticalSplitter": <div
    class="vertical-splitter"
  />,
  "workspace": <div
    class="silex-workspace"
  />,
}
`;
